Teradata/scripts/create_ddls.btq

```diff
@@ -1,25 +1,90 @@
+****    DDLS > 12K , Altered, comments with -- 
+**** Single export file TABLE and VIEWS (each)
+**** Triggers added to extraction process
+******-- OR T1.REQUESTTEXT like '%--%' OR T1.REQUESTTEXT like '%/*%'  	  -- Un comment this ine Comments are not needed 
+
 
 .LOGON connection_string;
 
+.EXPORT RESET
+
+.SET WIDTH 65531
+.set titledashes off
+
 **** CREATE TABLES FILE ****
+
+.EXPORT RESET
 .EXPORT FILE = ../temp/SHOW_Tables.sql
+
 .SET WIDTH 65531
-SELECT 'SELECT ''/* <sc-table> '' || ''' || TRIM(T1.DATABASENAME) || '.' || TRIM(T1.TABLENAME) || ' </sc-table> */'' as "--";     ' || 'SHOW TABLE ' || TRIM(T1.DATABASENAME) || '.' ||TRIM(T1.TABLENAME) || ';' "--" FROM DBC.TABLESV T1 WHERE T1.TABLEKIND IN ('T','O','Q') AND include_databases AND exclude_databases AND include_objects  GROUP BY 1; 
-.EXPORT RESET
+.set titledashes off
+LOCKING ROW  FOR ACCESS 
+-- GENERATE SHOW TABLE STEMENT FOR LONG DDLS
+SELECT 'SELECT ''/* <sc-table> '' || ''' || TRIM(DATABASENAME) || '.' || TRIM(TABLENAME) || ' </sc-table> */'' as "--";     ' || 'SHOW TABLE ' || TRIM(DATABASENAME) || '.' ||TRIM(TABLENAME) || ';' "--"
+ FROM 
+ (SELECT T1.DATABASENAME,  T1.TABLENAME,
+       T1.TABLEKIND,
+       CHARACTER_LENGTH(T1.REQUESTTEXT) DDL_LENG,
+       T1.REQUESTTEXT,
+       T1.CREATETIMESTAMP,
+       T1.LASTALTERTIMESTAMP,
+	T1.REQUESTTXTOVERFLOW,  -- MAX SIZE OF SQLTEXT IS 12500 
+    -- CREATE AND ALTER ARE SAME WHEN TABLE IS CREATED AND DIFFER IF ALTERED LATER 
+	CASE WHEN T1.CREATETIMESTAMP <> T1.LASTALTERTIMESTAMP 
+    OR  T1.REQUESTTXTOVERFLOW IS NOT NULL  OR DDL_LENG > 12400  -- MAX LIMIT IN DICTIONARY TABLE IS 12500
+        -- OBJECT ALTERED
+        OR  T1.REQUESTTEXT like '%ALTER%' THEN 'SHOW_DDL'     --  OBJECT ALTERED NEED SHOW OBJECT
+       	ELSE 'DICT_DDL' END AS STATUS  
+   FROM DBC.TABLESV T1
+   WHERE  T1.TABLEKIND IN ( 'T' ,'O','Q') -- PI AND NOPI 
+   AND include_databases AND exclude_databases AND include_objects
+   AND STATUS = 'SHOW_DDL' --LARGE DDL TABLES 
+   ) ASD ;
+
+
+
 .OS rm ../output/object_extracts/DDL/DDL_Tables.sql
 .EXPORT FILE = ../output/object_extracts/DDL/DDL_Tables.sql
-.SET WIDTH 65531
+
+LOCKING ROW  FOR ACCESS 
+-- GET TABLE DDL FROM DICTIONARY TABLE
+ SEL '/* <sc-table> ' ||  TRIM(DATABASENAME) ||'.'||TRIM(TABLENAME) || ' </sc-table> */' as "--", 
+TRIM(REQUESTTEXT) (TITLE'') , '--' (TITLE'') 
+FROM 
+ (SELECT T1.DATABASENAME,  T1.TABLENAME,
+       T1.TABLEKIND,
+       CHARACTER_LENGTH(T1.REQUESTTEXT) DDL_LENG,
+       T1.REQUESTTEXT,
+       T1.CREATETIMESTAMP,
+       T1.LASTALTERTIMESTAMP,
+	T1.REQUESTTXTOVERFLOW,  -- MAX SIZE OF SQLTEXT IS 12500 
+    -- CREATE AND ALTER ARE SAME WHEN TABLE IS CREATED AND DIFFER IF ALTERED LATER 
+	CASE WHEN T1.CREATETIMESTAMP <> T1.LASTALTERTIMESTAMP 
+   -- OR T1.REQUESTTEXT like '%--%' OR T1.REQUESTTEXT like '%/*%'  	-- UNCOMMENT THIS INE IF COMMENTS ARE NOT NEEDED
+    OR  T1.REQUESTTXTOVERFLOW IS NOT NULL  OR DDL_LENG > 12400  -- MAX LIMIT IN DICTIONARY TABLE IS 12500
```

Comment:
Is there a way to define a constant with this 12 400 value, with a representative name?

Reply:
We don't have an environment to test this files, I'll ask to Najeeb to do the change
Ty.